src/Aspire.Hosting/Redis/RedisBuilderExtensions.cs

```diff
@@ -19,33 +19,19 @@ public static class RedisBuilderExtensions
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <param name="port">The host port for the redis server.</param>
+    /// <param name="port">The host port to bind the underlying container to.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<RedisContainerResource> AddRedisContainer(this IDistributedApplicationBuilder builder, string name, int? port = null)
-    {
-        var redis = new RedisContainerResource(name);
-        return builder.AddResource(redis)
-                      .WithManifestPublishingCallback(context => WriteRedisContainerResourceToManifest(context, redis))
-                      .WithAnnotation(new EndpointAnnotation(ProtocolType.Tcp, port: port, containerPort: 6379))
-                      .WithAnnotation(new ContainerImageAnnotation { Image = "redis", Tag = "latest" });
-    }
-
-    /// <summary>
-    /// Adds a Redis container to the application model. The default image is "redis" and tag is "latest".
-    /// </summary>
-    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
-    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name)
+    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name, int? port = null)
```

Comment:
Allow specifying the image as well?

Reply:
I'm going to address that universally here: https://github.com/dotnet/aspire/issues/1997